### ISSUE TITLE
set cookie

### ISSUE THREAD

**Issue by lovemit** [hgourvest/dcef3#90]
How do I specify cookies and then go directly to the login success page?
Thank you very much

**Comment by PioneerDk** [comment]
Operating dcef3 directly with JS.

**Comment by lovemit** [comment]
@PioneerDk My current idea is to use idhttp to login to the web page, and then to display the Login successful page in cef3.Is it possible to use JS?

**Comment by PioneerDk** [comment]
No, you log in successfully and return the identification code. Let dcef3 access a success page with the successful ID code.

**Comment by lovemit** [comment]
@PioneerDk 
I don't know what you mean. I hope you can give me a DEMO code for reference.thank you
I have been using IDHTTP login cookies for a long time, and I will give the cookies to webbrowser to access the successful login page, but the Internet explorer browser is not very good for H5, so I want to use CEF in this project.

**Comment by a-sanders** [comment]
If you want to navigate using predefined cookies then you have to use TChromium.CookieManager before LoadUrl/LoadRequest:
Take a look at my code please:

```
Chromium.CookieManager.SetCookie(fURL,
        sCookieName, sCookieValue, sCookieDomain, sCookiePath, sCookieSecure,
        False, False, 0.0, Now, 0.0, nil);

Chromium.Browser.MainFrame.LoadUrl(fURL);
```

**Comment by lovemit** [comment]
@a-sanders 
I did this, but it prompted a memory error.
Do you have the successful case code for setcookie?
Can I refer to it?
Thank you very much!

**Comment by a-sanders** [comment]
Take a look at my gist "[How to set and delete cookies using CookieManager of CEF](https://gist.github.com/a-sanders/65d49fa78461fb07d0212b9f5578404b)"

**Comment by lovemit** [comment]
@a-sanders 
My friend, I don't know how to thank you, your code perfectly solved my problem.
Thank you very much for your help. I hope to have a better way to contact you and learn about Delphi technology.
